fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBInQueryTest.java

```diff
@@ -261,6 +267,77 @@ public void testInQueryIndexSortedDifferently() throws Exception {
                 context -> TestHelpers.assertDiscardedAtMost(40, context)));
     }
 
+    /**
+     * Verify that an IN query with a sort can be implemented as an ordered union of compound indexes that can satisfy
+     * the sort once the equality predicates from the IN have been pushed onto the indexes.
+     * @see com.apple.foundationdb.record.query.plan.planning.InExtractor#asOr()
+     */
+    @ParameterizedTest
+    @BooleanSource
+    public void inQueryWithSortBySecondFieldOfCompoundIndex(boolean shouldAttemptInAsOr) throws Exception {
+        RecordMetaDataHook hook = metaData ->
+                metaData.addIndex("MySimpleRecord", "compoundIndex",
+                        concat(field("num_value_3_indexed"), field("str_value_indexed")));
+        complexQuerySetup(hook);
+        final List<Integer> inList = asList(1, 4, 2);
+        RecordQuery query = RecordQuery.newBuilder()
+                .setRecordType("MySimpleRecord")
+                .setFilter(Query.field("num_value_3_indexed").in(inList))
+                .setSort(field("str_value_indexed"))
+                .build();
+
+        if (planner instanceof RecordQueryPlanner) { // will always be true, but it's good to check before casting
```

Comment:
I suppose technically, this could be a `@DualPlannerTest`, then (though not saying it should).

Or this could assert the planner type so that if someone were to make it a `DualPlannerTest` at some point, they get an error rather than it succeeding for the wrong reasons.

Reply:
I suppose it could. 🙃 

I switched this to an assertion instead.